### DISCUSSION TITLE
Dapp Testing help

### DISCUSSION THREAD

**Discussion by blkluv** [WalletConnect/walletconnect-monorepo#3430] [Developer Support]
Im not clear on how to test my dapp with https://react-web3wallet.vercel.app

"Dapp Testing - Could not verify if testing has been done on this dapp. As we do not have available V2 metric data for Sign. Please test your integration with our test wallet and submit for approval if successful. Link: https://react-web3wallet.vercel.app"

**Comment by EECvision** [comment]
> Im not clear on how to test my dapp with https://react-web3wallet.vercel.app
> 
> "Dapp Testing - Could not verify if testing has been done on this dapp. As we do not have available V2 metric data for Sign. Please test your integration with our test wallet and submit for approval if successful. Link: https://react-web3wallet.vercel.app"

I am having the same issue here. Please is anyone available to help?

**Reply by glitch-txs** [reply]
Is this error showing on Cloud website?

**Reply by EECvision** [reply]
Yes, here is the screenshot

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/63562663/5ab6299a-9bf1-45d6-bb5d-d8b3a0e1d55d)

**Reply by glitch-txs** [reply]
Thanks! This is to make sure your WalletConnect v2 integration works, we track these test Dapps/wallets to get information of your sessions before approving it. You need to test your wallet there before submitting. (e.g. connect + try transactions/signatures)

**Reply by EECvision** [reply]
Please can you provide more details? How is the link (https://lab.web3modal.com/) in the screenshot connected to my walletConnect integration?

Are my to test using the link?

If my wallet integration is not approved, will it prevent my app from connecting to mobile wallets?

**Reply by glitch-txs** [reply]
Hi, you don't need to get the approval in order to connect/work with WalletConnect, the verification is to be in the explorer API, this will allow your wallet to show in the modal

**Reply by EECvision** [reply]
Okay, that's the issue I'm having.
This is a sample wallet-connect integration I created: https://wallet-connect-iota-sable.vercel.app/

When I click on the "Connect wallet" button on my mobile phone, it takes me to metamask but does not trigger Connect wallet action. It remains idle instead. Please can you check the link to confirm?

**Reply by glitch-txs** [reply]
I see, do you have a public repo of this?

**Reply by EECvision** [reply]
Yes, I do. Here is the link: https://github.com/bleuwater2023startup/WalletConnect

**Reply by glitch-txs** [reply]
I'm able to connect with MetaMask on Android, which device are you testing with? Try upgrading the app to the latest version also MetMask is a bit slow currently

**Reply by EECvision** [reply]
Hi, it works fine now ☺. Thank you for your assistance.

**Reply by SardarShahzeb** [reply]
@glitch-txs i am facing this issue will you plz assest i connect my trust via uri on this link https://wallet-connect-iota-sable.vercel.app/.
and click on approve and it done successfully then i resubmit the draft but it get rejected again

**Reply by glitch-txs** [reply]
@SardarShahzeb submitions are for projects with original domains, they will be shown on our explorer. Also this is not a must for dapps, so you don't need to worry about it :)

**Reply by SardarShahzeb** [reply]
Thank you  @glitch-txs  for bringing this to my attention. In my DApp, there seems to be a significant amount of buying and user activity, leading to occasional errors with the verify API of wallet connect. After exploring various solutions, the issue was resolved when the project ID was changed with newone. It's possible that the root of the problem lies in the project not being verified and published. Would you like assistance in further investigating or resolving this?

**Reply by glitch-txs** [reply]
Make sure all versions are up to date for WalletConnect and Wagmi

**Reply by SardarShahzeb** [reply]
all updated using walletConnect with web3. i just want to confirm that project id once created. can be used and does it has any calls/sec issue or limit.

**Reply by glitch-txs** [reply]
You can use it right away, if you wish you can whitelist your domain to get CORS protection

**Reply by SardarShahzeb** [reply]
ok thanks really appreciated
